### PR TITLE
Control access permissions for a user in the directory server

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

File: src/main/java/org/gridsuite/gateway/GatewayConfig.java [hunk 1]
```diff
@@ -21,225 +18,88 @@
...
+
     @Bean
-    public RouteLocator myRoutes(RouteLocatorBuilder builder, UriConfiguration uriConfiguration) {
+    public RouteLocator myRoutes(RouteLocatorBuilder builder, ServicesURIsConfig servicesURIsConfig) {
         return builder.routes()
             .route(p -> p
-                    .path("/study/**")
-                    .filters(f -> f.rewritePath("/study/(.*)", "/$1"))
-                    .uri(uriConfiguration.getStudyServerBaseUri())
+                .path("/study/**")
```
Comment (JacquesBorsenberger) [inline src/main/java/org/gridsuite/gateway/GatewayConfig.java:29]: still sad that path("/" + "<PATTERN>" + "/**").uri( "/" + "<PATTERN>").uri(service.get...)

**Review by JacquesBorsenberger — CHANGES_REQUESTED** [review]
you should improve code  coverage in tests

File: src/main/java/org/gridsuite/gateway/ServiceURIsConfig.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+package org.gridsuite.gateway;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
```
Comment (JacquesBorsenberger) [inline src/main/java/org/gridsuite/gateway/ServiceURIsConfig.java:16]: are you shure?

File: src/test/java/org/gridsuite/gateway/ElementAccessControlTest.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            .withPort(port)
+            .willReturn(aResponse()
+                .withHeader("Content-Type", "application/json")));
+
+        stubFor(get(urlEqualTo(String.format("/v1/directories/%s/elements", uuid))).withHeader("userId", equalTo("user1"))
+            .willReturn(aResponse()
+                .withHeader("Content-Type", "application/json")
+                .withBody("[{\"name\": \"test\"}]")));
+
+//        webClient
```
Comment (JacquesBorsenberger) [inline src/test/java/org/gridsuite/gateway/ElementAccessControlTest.java:158]: comment in code

File: src/main/java/org/gridsuite/gateway/GatewayConfig.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/gateway/GatewayConfig.java]: Done

File: src/main/java/org/gridsuite/gateway/ServiceURIsConfig.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/gateway/ServiceURIsConfig.java:16]: yes

File: src/main/java/org/gridsuite/gateway/endpoints/CaseServer.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*
+  Copyright (c) 2021, RTE (http://www.rte-france.com)
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/gateway/endpoints/CaseServer.java:2]: all copyright from new files should be 2022

**Comment by AbdelHedhili** [comment]
If possible increase a bit code coverage (if it doesn't imply doing dull tests)

File: src/main/java/org/gridsuite/gateway/endpoints/CaseServer.java (hunk 4 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/gateway/endpoints/CaseServer.java:2]: Created in 2021 but pushed to 2022

File: src/test/java/org/gridsuite/gateway/ElementAccessControlTest.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/gateway/ElementAccessControlTest.java]: Removed

**Review by AbdelHedhili — COMMENTED** [review]
Copyrights from new files should be 2022

**Review by AbdelHedhili — APPROVED** [review]

**Review by JacquesBorsenberger — APPROVED** [review]

**Pull request merged**